### ISSUE TITLE
Can't acces method in Controller

### ISSUE THREAD

**Issue by zahidmuhammadzaki** [fmalk/codeigniter-phpunit#40]
I have this function in Welcome Controller

```
public function format_tanggal($tanggal)
{
 return date('d-m-Y',strtotime($tanggal));
}
```

But when i write test in WelcomeTest

```
public function testFormatTanggal(){
        $this->requireController('Welcome');
        $this->CIC = new Welcome();
        $this->assertInstanceOf('CI_Controller', $this->CIC, 'Controller was called correctly');
        $this->assertEquals($this->CIC->format_tanggal('02-05-2015'),'05-02-2015');
}
```

I get this error

![error](https://cloud.githubusercontent.com/assets/8657754/12838185/66c9dfe4-cbfd-11e5-93f6-ad26fa68a0a6.JPG)

**Comment by fmalk** [comment]
Should be easy to XDebug it.
Your `$this->CIC` must not have been initialized with the Welcome Controller you want.
I also checked CI docs for your method name (that underscore raised some flags for me), but it doesn't seem to be a reserved word since it doesn't actually begins with an underscore.

**Comment by zahidmuhammadzaki** [comment]
Thank's, it solved. But i have another question, is it support HMVC?

**Comment by fmalk** [comment]
@zahidmuhammadzaki you mean HHVM? Yes it does.
I have a Travis CI on it: https://travis-ci.org/fmalk/codeigniter-phpunit/jobs/99513092

**Comment by zahidmuhammadzaki** [comment]
No, I mean Hierarchical Model View Controller, it's modular method in CI.

**Comment by fmalk** [comment]
Oh, sorry.
No idea. It puts files in core folder so it'd have to be tested. I honestly don't know if it is a popular extension.

**Comment by zahidmuhammadzaki** [comment]
Yeah, it is popular extension.. I hope it can be included in the next development..
Thanks for your response..

**Comment by fmalk** [comment]
I'll consider compatibility within a next minor release, so I'll reopen this issue as a "feature request"

**Comment by fmalk** [comment]
@zahidmuhammadzaki don't forget to comment here if you find any success or bugs trying HMVC with this project

**Comment by zahidmuhammadzaki** [comment]
Yes, i will..

**Comment by zahidmuhammadzaki** [comment]
I found a hack solution for this HMVC problem..

But firstly, i want to describe what i use and i do:

1. I use wiredesignz [HMVC](https://bitbucket.org/wiredesignz/codeigniter-modular-extensions-hmvc)
The folders structure looks like this
![1](https://cloud.githubusercontent.com/assets/8657754/13044797/c0b4a45a-d402-11e5-8e68-09f4e6ff454c.JPG)

2. I add this function to CITestCase.php to get controller from modules folders
```
	public function requireModuleController($module_path, $prefix = null)
	{
		list($module,$class) = explode('/', $module_path);
		$filepath = APPPATH.'modules'.DIRECTORY_SEPARATOR.$module.DIRECTORY_SEPARATOR.'controllers'.DIRECTORY_SEPARATOR.$prefix.DIRECTORY_SEPARATOR.$class.'.php';
		if (file_exists($filepath))
		{
			require_once($filepath);
		}
	}
```

The problem is when i use this function i get this error
![error2](https://cloud.githubusercontent.com/assets/8657754/13045694/6f35e864-d407-11e5-81f8-99f0a7a00da7.JPG)

So, i debug the function in the Modules.php
![error3](https://cloud.githubusercontent.com/assets/8657754/13045744/b842fa92-d407-11e5-8250-99eb21f8d1c6.JPG)
And i get NULL value in phpunit

But i get CI_Config Instance when i run in browser
![error4](https://cloud.githubusercontent.com/assets/8657754/13045833/2c15346c-d408-11e5-96b8-9dd7400c1ed6.JPG)

It seem global $CFG is not work in phpunit, i have no idea..

I found some solution to prevent phpunit backups all global variables.
So, I try to disable php backupGlobals to "false"
![error6](https://cloud.githubusercontent.com/assets/8657754/13046120/a326b778-d409-11e5-9b27-30dc3de5f8db.JPG)
But it does not work, and i still get null value in phpunit

Then i add few of code to Modules.php
![error5](https://cloud.githubusercontent.com/assets/8657754/13045940/a52a4180-d408-11e5-859c-f1cbc646bb58.JPG)

And, It's works!

I want to make it works without hack the HMVC source file.
Can anyone help?
